happi/utils.py

```diff
@@ -118,3 +119,21 @@ def warn_and_call(*args, **kwargs):
         return warn_and_call
 
     return wrapper
+
+
+def build_abs_path(basedir: str, path: str) -> str:
```

Comment:
I like this utils function :+1:, its definitely on the long long list of "why isn't this in the standard library"

Reply:
I've been using pathlib for a while and they support it: [p.resolve()](https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve)

Reply:
I don't see any function in the standard library for prepending a base directory if the tail directory is not absolute though

Reply:
Ah I see what you mean.  That is inconvenient